Source/control.cpp

```diff
@@ -370,7 +370,7 @@ void RemoveGold(Player &player, int goldIndex)
 	else
 		player.RemoveInvItem(gi);
 	InitializeItem(player.HoldItem, IDI_GOLD);

```

Comment:
You've removed every use of `InitializeItem(..., IDI_GOLD)` except for this one. TBH my preference would be to keep these calls instead of copying from `GoldTemplate` but consistency one way or the other would be better than mixing.

Reply:
I prefer that my self, but Constancy was most important part and I wast sure if there would be a performance benefit. So wanted some feedback first :)

Reply:
Alright I have made a helper function for generating gold stacks, I think this is both cleaner and probably even more optimal performance wise then copying the template.